### ISSUE TITLE
Support jdk and aot proxies without a root @NativeHint

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-native#830]

**Comment by aclement** [comment]
I can't reproduce this. In my testing in the project and with Ollies restbucks sample I see the proxies in proxy-config.json with/without the root NativeHint. What I did:
```
git clone https://github.com/odrotbohm/spring-restbucks
cd spring-restbucks
git checkout hacking/native
cd server
mvn spring-boot:build-image &> log.txt
```
The command fails for me but I can look at the generated `proxy-config.json`. 

I then played with the file in question `NativeConfiguration.java` which has this:
```
@NativeHint(
  jdkProxies = @JdkProxyHint(types = { TransactionalEventListener.class, SynthesizedAnnotation.class })
)
```
To avoid confusing that with what we are doing for you around TransactionalEventListener, I tweaked the types involved to List/Serializable.

I could see it in proxy-config.json after a build (mvn spring-boot:build-image), I removed the hint completely it disappeared then I re-did it as a top level hint (not in a NativeHint) and it reappeared. I removed all the other hints to see if that impacted it, then re-added them. At each stage it was as expected (The unit tests for this seem to be behaving too)

I don't deny there may be an issue but I'm having trouble creating it. These horrible build times make it hard to try lots of things...
